platform/ios/src/MGLScaleBarView.m

```diff
@@ -0,0 +1,289 @@
+#import <Mapbox/Mapbox.h>
+
+#import "MGLScaleBarView_Private.h"
+
+@interface MGLScaleBarView()
+@property (nonatomic, assign, getter=isVisible) BOOL visible;
+@property (nonatomic, assign) CLLocationDistance metersPerPoint;
+@property (nonatomic) NSArray<UILabel *> *labels;
+@property (nonatomic) NSArray<UIView *> *bars;
+@property (nonatomic) UIView *containerView;
+@property (nonatomic) MGLDistanceFormatter *formatter;
+@property (nonatomic) NSArray<NSArray<NSNumber*>*> *metricTable;
+@property (nonatomic) NSArray<NSNumber*> *row;
+@property (nonatomic) UIColor *primaryColor;
+@property (nonatomic) UIColor *secondaryColor;
+@property (nonatomic) CALayer *borderLayer;
+@end
+
+static const CGFloat BAR_HEIGHT = 4;
+
+@implementation MGLScaleBarView
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    if (self = [super initWithCoder:aDecoder]) {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    if (self = [super initWithFrame:frame]) {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (void)commonInit {
+    self.clipsToBounds = NO;
+    self.alpha = 0;
+    
+    _visible = NO;
+    _primaryColor = [UIColor colorWithRed:62.0/255.0 green:62.0/255.0 blue:62.0/255.0 alpha:1];
+    _secondaryColor = [UIColor colorWithRed:247.0/255.0 green:247.0/255.0 blue:247.0/255.0 alpha:1];
```

Comment:
The text and border colors should switch between white-on-black and black-on-white, depending on the background color of the current map style — e.g., the satellite and dark styles would need colors that are inverted from the current black-on-white.